core/trino-main/src/main/java/io/trino/operator/OperatorFactories.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator;
+
+import io.trino.spi.type.Type;
+import io.trino.spiller.PartitioningSpillerFactory;
+import io.trino.sql.planner.plan.PlanNodeId;
+import io.trino.type.BlockTypeOperators;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.OptionalInt;
+
+public interface OperatorFactories
```

Comment:
For me the `OperatorFactories` are a bit misleading, maybe `JoinOperatorFactories` - since it involves only joins

Reply:
I expect this will be extended with other operators too

Reply:
renamed to `TrinoOperatorFactories`